src/Rules/UnusedPublicClassConstRule.php

```diff
@@ -105,7 +107,6 @@ private function isClassConstantUsed(
 
         $publicConstantReference = $className . '::' . $constantName;
 
-        $usedConstFetches = Arrays::flatten($usedConstFetches);
```

Comment:
the PR only moves this flatten call before the nested loops, so we do the work only once, instead of on every iteration.

the input parameters are already known before the loops